api/v1/submissions/PKPSubmissionHandler.php

```diff
@@ -665,10 +665,27 @@ public function submit(SlimRequest $slimRequest, APIResponse $response, array $a
     {
         $request = $this->getRequest();
         $context = $request->getContext();
-        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION); /** @param Submission $submission */
```

Comment:
Here it should be `@var`, other alternatives:

- After L668:
```php
assert($submissionKeywordEntryDao instanceof SubmissionKeywordEntryDAO);
```
- Before L668 (without the variable name):
```php
/** @var Submission */
```

I think the assert is great for such methods with a dynamic return :)

Reply:
Thanks! I have a vague memory of Alec saying he wanted to get rid of all of the `asset()`s in the codebase, so I've gone with `/** @var` for now.

Reply:
Assertions have sometimes gotten us into hot water because they run disabled on many systems (including some of the dev team). I'd rather we replaced them with type hinting and exceptions, except in cases where an assert for some reason is really the best tool.

Reply:
I was just thinking about using more assertive type hints when possible. As you can see below, the `instanceof` and `assert()` have the doctype power (at least when using a popular vscode extension 👀).

![image](https://user-images.githubusercontent.com/361921/208850775-48e0293b-0c4b-483f-ab9f-d1c937206b2b.png)